pkg/sql/sqlbase/testutils.go

```diff
@@ -1473,6 +1552,103 @@ func randIndexTableDefFromCols(
 	return tree.IndexTableDef{Columns: indexElemList}
 }
 
+// randPartialIndexPredicateFromCols creates a partial index expression with
+// a random subset of the given columns.
+func randPartialIndexPredicateFromCols(
+	rng *rand.Rand, columnTableDefs []*tree.ColumnTableDef, tableName *tree.TableName,
+) tree.Expr {
+	// Shuffle the columns.
+	cpy := make([]*tree.ColumnTableDef, len(columnTableDefs))
+	copy(cpy, columnTableDefs)
+	rng.Shuffle(len(cpy), func(i, j int) { cpy[i], cpy[j] = cpy[j], cpy[i] })
+
+	// Select a random number of columns (at least 1). Loop through the columns
+	// to find columns with types that are currently supported for generating
+	// partial index expressions.
+	nCols := rng.Intn(len(cpy)) + 1
+	cols := make([]*tree.ColumnTableDef, 0, nCols)
+	for _, col := range cpy {
+		if isAllowedPartialIndexColType(col) {
+			cols = append(cols, col)
+		}
+		if len(cols) == nCols {
+			break
+		}
+	}
+
+	// Build a boolean expression tree with containing a reference to each
+	// column.
+	var e tree.Expr
```

Comment:
It looks like it's possible that none of the column types are are useable, in which case e will be nil.

Reply:
This should be fine, because a `nil` `Predicate` indicates a non-partial index. I guess it may be confusing that this function returns nil when called, but it shouldn't break anything. I've added a comment to the function to call this out, but if there's something else you think is more appropriate, I'm all ears.